osquery/tables/events/darwin/yara.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ *  Copyright (c) 2015, Wesley Shields
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <map>
+#include <string>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+
+#include <yara.h>
+
+#include "osquery/events/darwin/fsevents.h"
+
+namespace osquery {
+namespace tables {
+
+/**
+ * @brief Track YARA matches to files.
+ */
+class YARAEventSubscriber : public EventSubscriber<FSEventsEventPublisher> {
+  DECLARE_SUBSCRIBER("yara");
+
+ public:
+  void init();
+
+ private:
+  // XXX: Is there a better way to say "I'm not ready to receive events"?
```

Comment:
I guess my comment wasn't very clear. I use that boolean in the event handler and quickly return if we aren't "ready" yet. This is because events still fire because there is no way to say "don't give me events, I've got a problem!" It seems like init() should have some way of indicating that it was successful or not. If not successful the subscriber should be unsubscribed.